src/hooks/useAppDetails.ts

```diff
@@ -0,0 +1,118 @@
+import { useQuery } from "@tanstack/react-query";
+import { ethers } from "ethers";
+import { NetworkId } from "src/constants";
+import { SOHM_CONTRACT, STAKING_CONTRACT, V1_STAKING_CONTRACT } from "src/constants/contracts";
+import { queryAssertion } from "src/helpers/react-query/queryAssertion";
+import {
+  useMarketCap,
+  useOhmCirculatingSupply,
+  useOhmPrice,
+  useOhmTotalSupply,
+  useTotalValueDeposited,
+  useTreasuryMarketValue,
+} from "src/hooks/useProtocolMetrics";
+import { useProvider } from "wagmi";
+
+export interface IAppData {
+  readonly circSupply?: number;
+  readonly currentIndex?: string;
+  readonly currentIndexV1?: string;
+  readonly currentBlock?: number;
+  readonly fiveDayRate?: number;
+  readonly loading: boolean;
+  readonly loadingMarketPrice: boolean;
+  readonly marketCap?: number;
+  readonly marketPrice?: number;
+  readonly stakingAPY?: number;
+  readonly stakingRebase?: number;
+  readonly stakingTVL?: number;
+  readonly totalSupply?: number;
+  readonly treasuryBalance?: number;
+  readonly treasuryMarketValue?: number;
+  readonly secondsToEpoch?: number;
+}
+
+export const appDetailsKey = () => ["useAppDetails"];
+
+export const useAppDetails = (subgraphUrl?: string) => {
```

Comment:
would we prefer to remove this big hook and instead just use the smaller hooks where the consumers need it?

Reply:
Up for suggestions for a better name than useAppDetails